cypress/support/api/utils/discounts/salesUtils.js

```diff
@@ -15,42 +16,61 @@ export function createSaleInChannel({
   type,
   value,
   channelId,
-  discountValue = value
+  discountValue = value,
 }) {
   let sale;
   return createSale({
     name,
     type,
-    value
+    value,
   })
     .then(saleResp => {
       sale = saleResp;
       addChannelToSale(sale.id, channelId, discountValue);
     })
     .then(() => sale);
 }
+export function shouldVariantBeOnSale(
+  variantId,
+  channelSlug,
+  shouldBeSales,
+  retries = 0,
+) {
+  return getVariant(variantId, channelSlug).then(salesResponse => {
+    if (salesResponse.pricing.onSale === shouldBeSales) {
+      return;
```

Comment:
**question**: you don't return anything here but you chain cypress `.its` to response?

Reply:
This retrun is to break "loop" and I retrun response from 39 line

Reply:
This retrun is to break "loop" and I retrun response from 39 line